tools/travis/lnx-cibuild.cmake

```diff
@@ -15,13 +15,14 @@ message(STATUS "CTEST_BINARY_DIRECTORY: ${CTEST_BINARY_DIRECTORY}")
 set(INITIAL_CACHE
 "CMAKE_PREFIX_PATH=$ENV{SOURCE_DIRECTORY}/contrib\;/usr
 BOOST_USE_STATIC=Off
-CMAKE_BUILD_TYPE=Release
+CMAKE_BUILD_TYPE=$ENV{CMAKE_BUILD_TYPE}
```

Comment:
it seems that before, travis built in release mode anyways and ignored the variable introduced in #1578